consensus/src/liveness/cached_proposer_election.rs

```diff
@@ -22,7 +22,7 @@ pub struct CachedProposerElection {
     // to look back (and caller knows how big of a window it needs).
     // LRU cache wouldn't work as well, as access order of the elements
     // would define eviction, and could lead to evicting still needed elements.
-    recent_elections: Mutex<BTreeMap<Round, Author>>,
+    recent_elections: Mutex<BTreeMap<Round, (Author, f64)>>,
```

Comment:
nit: ^ above, there is `Additionally, provides is_valid_proposal that remembers,` but it seems to not be true (anymore)?